src/ReverseProxy/Health/EntityActionScheduler.cs

```diff
@@ -24,62 +25,80 @@ namespace Yarp.ReverseProxy.Health
     internal sealed class EntityActionScheduler<T> : IDisposable where T : notnull
     {
         private readonly ConcurrentDictionary<T, SchedulerEntry> _entries = new();
+        private readonly WeakReference<EntityActionScheduler<T>> _weakThisRef;
         private readonly Func<T, Task> _action;
         private readonly bool _runOnce;
         private readonly ITimerFactory _timerFactory;
-        private readonly TimerCallback _timerCallback;
-        private int _isStarted;
+
+        private const int Status_NotStarted = 0;
+        private const int Status_Started = 1;
+        private const int Status_Disposed = 2;
```

Comment:
Naming seems strange because I've not seen previously us using `_` in the middle on member names in production code.
```suggestion
        private const int NotStarted = 0;
        private const int Started = 1;
        private const int Disposed = 2;
```

Reply:
We've used it in a few places in [runtime](https://github.com/dotnet/runtime/blob/3d5d26181cd7bc07ea6c6f87710c14ccd043b415/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L71-L73), but I get it's not common. I'm fine with renaming them.